### PR TITLE
DBZ-2650 Migrated release related jobs from JJB to JobDSL

### PR DESCRIPTION
**Pull request by jcechace**

### CODE REVIEW

**Comment by jcechace** [comment]
@jpechane  This will need a thorough. Before it gets merged I will also import this manually into our new Jenkins instance so that these jobs can be tested -- which I will have to delegate upon you.

File: jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        stringParam('ARTIFACT_DIR', '/mnt/rcm-guest/staging/amq', 'Staging directory')
+    }
+
+    definition {
+        cps {
+            script(readFileFromWorkspace('jenkins-jobs/pipelines/prod-release-pipeline.groovy'))
+        }
+    }
+
+    publishers {
```
Comment (jcechace) [inline jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy:38]: this needs to be moved into the pipeline as pipeline job in job DSL does not have publisher

**Comment by gunnarmorling** [comment]
@jpechane, any comments?

**Comment by jcechace** [comment]
> @jpechane, any comments?

We had a talk about this over hangouts. should be ready for testing tomorrow morning

**Comment by jcechace** [comment]
@jpechane should be ready for review and tests. Mainly the release pipelines should be tested

**Comment by jcechace** [comment]
Any update on this @jpechane ?

File: jenkins-jobs/job-dsl/release_debezium_nightly_image.groovy [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    wrappers {
+        timeout {
+            noActivity(5)
+        }
+    }
+
+    triggers {
+        upstream('debezium-deploy-snapshots')
```
Comment (jpechane) [inline jenkins-jobs/job-dsl/release_debezium_nightly_image.groovy:24]: Refers to renamed job

File: jenkins-jobs/job-dsl/tools_source_clear.groovy [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        githubProjectUrl('https://github.com/debezium/debezium')
+    }
+
+    logRotator {
+        daysToKeep(7)
+    }
+
+    wrappers {
+        timeout {
+            noActivity(90)
```
Comment (jpechane) [inline jenkins-jobs/job-dsl/tools_source_clear.groovy:19]: Needs larger timeout

**Comment by jpechane** [comment]
All delivered email looks like
`Run https://master-jenkins-csb-debezium.cloud.paas.psi.redhat.com/job/release-debezium-upstream/2/ finished with result: null` - why is there null?

Also could we have from set to something else than `address not configured yet`?
`

File: jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    logRotator {
+        numToKeep(5)
+    }
+
+    parameters {
+        stringParam('BUILD_VERSION', '', 'Maven artifact id of the product binaries')
+        stringParam('BUILD_VERSION_INCUBATOR', '', 'Maven artifact id of the product binaries in incubator')
+        stringParam('PRODUCT_VERSION', '', 'Product version')
+        stringParam('USERNAME', '', 'Username to log to staging host')
+        stringParam('PASSWORD', 'master', 'Password to log to staging host')
```
Comment (jpechane) [inline jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy:18]: Should be password, not string

File: jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        stringParam('CONNECTORS', 'mysql postgres mongodb sqlserver', 'The list of released connectors')
+        stringParam('CONNECTORS_INCUBATOR', 'db2', 'The list of released incubating connectors')
+        stringParam('STAGE_FILES', 'false', 'When checked the uploaded artifacts are staged')
```
Comment (jpechane) [inline jenkins-jobs/job-dsl/release_debezium_stage_prod.groovy:21]: Should be checkbox

**Comment by jcechace** [comment]
> All delivered email looks like
> `Run https://master-jenkins-csb-debezium.cloud.paas.psi.redhat.com/job/release-debezium-upstream/2/ finished with result: null` - why is there null?
> 
> Also could we have from set to something else than `address not configured yet`?
> `

Will fix. The address fix is not part of this PR (it's a Jenkins config).

**Comment by jcechace** [comment]
@jpechane  Everything should be resolved, I think we are done?

**Comment by jpechane** [comment]
@jcechace Yes, just please make sure your moved pipelines are synced with those in master, see for example
`git diff -w master:jenkins-jobs/scripts/release-pipeline.groovy jcechace-DBZ-2650:jenkins-jobs/pipelines/release-pipeline.groovy` - just use your branch names

**Comment by jpechane** [comment]
@jcechace Applied, thanks a lot!

**Pull request merged**